teamA/crud-express/heeyeon/app.js

```diff
@@ -0,0 +1,178 @@
+require('dotenv').config();
+
+const http = require('http');
+
+const express = require('express');
+const cors = require('cors');
+const morgan = require('morgan');
+const { DataSource } = require('typeorm');
+
+const myDataSource = new DataSource({
+    type: process.env.TYPEORM_CONNECTION,
+    host: process.env.TYPEORM_HOST,
+    port: process.env.TYPOPRM_PORT,
+    username: process.env.TYPEORM_USERNAME,
+    password: process.env.TYPEORM_PASSWORD,
+    database: process.env.TYPEORM_DATABASE
+})
+
+myDataSource.initialize()
+    .then(() => {
+        console.log('Data Source has been initialized!')
+    });
+
+const app = express()
+
+app.use(express.json());
+app.use(morgan('tiny'));
+
+// health check
+app.get('/ping', cors(), function (req, res, next) {
+    res.status(200).json({ message : 'pong '})
+});
+
+// 4. 전체 게시글 조회하기
+app.get('/posts/lists', async (req, res, next) => {
+
+    await myDataSource.query(
+        `SELECT
+            p.user_id As userId,
+            u.profile_image As userProfileImage,
+            p.id AS postingId,
+            p.posting_img_url As postingImageUrl,
+            p.content As postingContent
+        FROM users u, posts p WHERE p.user_id=u.id;`
+    ,(err, rows) => {
+            res.status(200).json({ "data" : rows });
+    })
+});
+
+// 5. 유저의 게시글 조회하기
+app.get('/posts/:userId', async (req, res, next) => {
+    const userId = req.params.userId;
+
+    const user = await myDataSource.query(
+        `SELECT
+            id AS userId,
+            profile_image AS userProfileImage
+        FROM users WHERE id=${userId};`
+    );
+
+    const post = await myDataSource.query(
+        `SELECT
+            id AS postingId,
+            posting_img_url AS postingImageUrl,
+            content AS postingContent
+        FROM posts WHERE user_id=${userId};`
+    );

```

Comment:
유저의 정보 (user_id)를 통해 특정 게시물을 검색하는 방법은 다음과 같이 리팩토링 할 수 있습니다.

```suggestion
    const userId = req.params.userId;

    const getPostByUserId = await myDataSource.query(
        `SELECT
            posts.id AS postingId,
            posts.posting_img_url AS postingImageUrl,
            posts.content AS postingContent 
        FROM posts
        WHERE posts.user_id = ?`, [userId]
    );

```

Reply:
10/13(목) 예정 질문#2
req에 포함되는 userId이 들어갈 자리를 ?로 처리하는 쿼리 컨벤션은 이해됩니다.
69-74라인도 삭제해야 하는지 궁금합니다